libbpf-cargo/src/test.rs

```diff
@@ -518,7 +521,96 @@ fn test_skeleton_basic() {
     "#}
     .to_string();
 
-    let () = build_rust_project_from_bpf_c(&bpf_c, &rust);
+    build_rust_project_from_bpf_c(&bpf_c, &rust);
+}
+
+#[test]
+fn test_skeleton_generate_bpf_objs_section() {
+    let bpf_c = indoc! {r#"
+        #include "vmlinux.h"
+        #include <bpf/bpf_helpers.h>
+
+        struct {
+                __uint(type, BPF_MAP_TYPE_HASH);
+                __uint(max_entries, 1024);
+                __type(key, u32);
+                __type(value, u64);
+        } mymap SEC(".maps");
+
+        SEC("kprobe/foo")
+        int this_is_my_prog(u64 *ctx)
+        {
+                return 0;
+        }
+    "#}
+    .to_string();
+    let rust = indoc! {r#"
+        #![warn(elided_lifetimes_in_paths)]
+        mod bpf;
+        use std::mem::MaybeUninit;
+        use bpf::*;
+        use libbpf_rs::skel::SkelBuilder;
+        use libbpf_rs::skel::OpenSkel;
+        use libbpf_rs::skel::Skel;
+
+        fn main() {
+            let builder = ProgSkelBuilder::default();
+            let mut open_object = MaybeUninit::uninit();
+            let open_skel = builder
+                .open(&mut open_object)
+                .expect("failed to open skel");
+
+            // Check that we can grab handles to open maps/progs
+            let _open_map = &open_skel.maps.mymap;
+            let _open_prog = &open_skel.progs.this_is_my_prog;
+
+            let mut skel = open_skel.load().expect("failed to load skel");
+
+            // Check that we can grab handles to loaded maps/progs
+            let _map = &skel.maps.mymap;
+            let _prog = &skel.progs.this_is_my_prog;
+
+            // Check that attach() is generated
+            skel.attach().expect("failed to attach progs");
+
+            // Check that Option<Link> field is generated
+            let _mylink = skel.links.this_is_my_prog.unwrap();
```

Comment:
Do we really need all these supposed checks here? Looks more like copy&paste remnants to me.
```suggestion
            let mut open_object = MaybeUninit::uninit();
            let open_skel = builder
                .open(&mut open_object)
                .expect("failed to open skel");
            let _skel = open_skel.load().expect("failed to load skel");
```

Reply:
Removed as suggested, and also removed most of the BPF-C program that isn't necessary for the test.